airflow/providers/http/hooks/http.py

```diff
@@ -246,3 +253,142 @@ def test_connection(self):
+            else:
+                # schema defaults to HTTP
+                schema = conn.schema if conn.schema else "http"
+                host = conn.host if conn.host else ""
+                self.base_url = schema + "://" + host
+
+            if conn.port:
+                self.base_url = self.base_url + ":" + str(conn.port)
+            if conn.login:
+                auth = self.auth_type(conn.login, conn.password)
+            if conn.extra:
+                try:
+                    _headers.update(conn.extra_dejson)
+                except TypeError:
+                    self.log.warning("Connection to %s has invalid extra field.", conn.host)
+        if headers:
+            _headers.update(headers)
+
+        if self.base_url and not self.base_url.endswith("/") and endpoint and not endpoint.startswith("/"):
+            url = self.base_url + "/" + endpoint
+        else:
+            url = (self.base_url or "") + (endpoint or "")
+
+        async with aiohttp.ClientSession() as session:
+            if self.method == "GET":
+                request_func = session.get
+            elif self.method == "POST":
+                request_func = session.post
+            elif self.method == "PATCH":
+                request_func = session.patch
+            else:
+                raise AirflowException(f"Unexpected HTTP Method: {self.method}")
+
+            attempt_num = 1
+            while True:
+                response = await request_func(
+                    url,
+                    json=data if self.method in ("POST", "PATCH") else None,
+                    params=data if self.method == "GET" else None,
+                    headers=headers,
+                    auth=auth,
+                    **extra_options,
+                )
+                try:
+                    response.raise_for_status()
+                    return response
+                except ClientResponseError as e:
+                    self.log.warning(
+                        "[Try %d of %d] Request to %s failed.",
+                        attempt_num,
+                        self.retry_limit,
+                        url,
+                    )
+                    if not self._retryable_error_async(e) or attempt_num == self.retry_limit:
+                        self.log.exception("HTTP error with status: %s", e.status)
+                        # In this case, the user probably made a mistake.
+                        # Don't retry.
+                        raise AirflowException(str(e.status) + ":" + e.message)
+
+                attempt_num += 1
+                await asyncio.sleep(self.retry_delay)
+
+    def _retryable_error_async(self, exception: ClientResponseError) -> bool:
+        """
+        Determines whether or not an exception that was thrown might be successful
+        on a subsequent attempt.
+
+        It considers the following to be retryable:
+            - requests_exceptions.ConnectionError
+            - requests_exceptions.Timeout
```

Comment:
I can't find place where we catch this exceptions.

Reply:
For timeout and connection error, it would be <500